cmd/prober/prober.go

```diff
@@ -104,45 +107,119 @@ func encodeLevel() zapcore.LevelEncoder {
+	flag.BoolVar(&oneTime, "one-time", false, "Whether to run only one time and exit")
+	flag.BoolVar(&runWriteProber, "write-prober", false, " [Kubernetes only] run the probers for the write endpoints")
 
-	flag.StringVar(&tsaURL, "tsa-url", "https://timestamp.sigstore.dev", "Set to the TSA URL to run probers against")
+	flag.StringVar(&rekorRequestsJSON, "rekor-requests", "[]", "Additional rekor requests (JSON array)")
 
-	flag.BoolVar(&oneTime, "one-time", false, "Whether to run only one time and exit.")
-	flag.BoolVar(&runWriteProber, "write-prober", false, " [Kubernetes only] run the probers for the write endpoints.")
+	flag.StringVar(&fulcioRequestsJSON, "fulcio-requests", "[]", "Additional fulcio requests (JSON array)")
 
-	var rekorRequestsJSON string
-	flag.StringVar(&rekorRequestsJSON, "rekor-requests", "[]", "Additional rekor requests (JSON array.)")
+	flag.UintVar(&retries, "retry", 4, "Maximum number of retries before marking HTTP request as failed")
 
-	var fulcioRequestsJSON string
-	flag.StringVar(&fulcioRequestsJSON, "fulcio-requests", "[]", "Additional fulcio requests (JSON array.)")
-
-	flag.UintVar(&retries, "retry", 4, "maximum number of retries before marking HTTP request as failed")
+	flag.IntVar(&grpcPort, "grpc-port", 5554, "Port for Fulcio gRPC endpoint (only if configured)")
 
 	flag.Parse()
 
+	switch {
+	case scPath != "" && trPath != "":
+		signingConfig, err = root.NewSigningConfigFromPath(scPath)
+		if err != nil {
+			log.Fatal("Failed to load signing config: ", err)
+		}
+		trustedRoot, err = root.NewTrustedRootFromPath(trPath)
+		if err != nil {
+			log.Fatal("Failed to load trusted root: ", err)
+		}
+	case scPath == "" && trPath == "":
+		if staging {
+			opts := tuf.DefaultOptions()
+			opts.Root = tuf.StagingRoot()
+			opts.RepositoryBaseURL = tuf.StagingMirror
+			signingConfig, err = root.FetchSigningConfigWithOptions(opts)
+			if err != nil {
+				log.Fatal("Failed to fetch staging signing config: ", err)
+			}
+			trustedRoot, err = root.FetchTrustedRootWithOptions(opts)
+			if err != nil {
+				log.Fatal("Failed to fetch staging trusted root: ", err)
+			}
+		} else {
+			signingConfig, err = root.FetchSigningConfig()
+			if err != nil {
+				log.Fatal("Failed to fetch prod signing config: ", err)
+			}
+			trustedRoot, err = root.FetchTrustedRoot()
+			if err != nil {
+				log.Fatal("Failed to fetch prod trusted root: ", err)
+			}
+		}
+	default:
+		log.Fatal("Must specify both --signing-config and --trusted-root, or neither")
+	}
+
+	rekorV1Services, err = root.SelectServices(signingConfig.RekorLogURLs(), root.ServiceConfiguration{Selector: prototrustroot.ServiceSelector_ALL}, []uint32{1}, time.Now())
+	if err != nil {
+		log.Fatal("Failed to select Rekor services: ", err)
+	}
+	rekorV1URLs = mapServicesToURLs(rekorV1Services)
+
+	fulcioService, err = root.SelectService(signingConfig.FulcioCertificateAuthorityURLs(), []uint32{1}, time.Now())
```

Comment:
You can also use https://github.com/sigstore/sigstore-go/blob/070130677531d3e49c84866bee26d43ea710d2ac/pkg/sign/certificate.go#L61 to specify the version, I've exported variables for the supported versions for each service.